server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java

```diff
@@ -101,4 +110,11 @@ public boolean equals(@Nullable Object o) {
   public int hashCode() {
     return state.hashCode();
   }
+
+  public interface LifecycleListener {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Annotate the "LifecycleListener" interface with the @FunctionalInterface annotation [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1609)